packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -112,4 +125,39 @@ export class EVMSmartWallet {
     public async nfts() {
         return this.crossmintService.fetchNFTs(this.address, this.chain);
     }
+
+    public async sendTransaction<
+        const TAbi extends Abi | readonly unknown[],
+        TFunctionName extends ContractFunctionName<TAbi, "nonpayable" | "payable"> = ContractFunctionName<
+            TAbi,
+            "nonpayable" | "payable"
+        >,
+        TArgs extends ContractFunctionArgs<TAbi, "nonpayable" | "payable", TFunctionName> = ContractFunctionArgs<
+            TAbi,
+            "nonpayable" | "payable",
+            TFunctionName
+        >
+    >({
+        address,
+        abi,
```

Comment:
What does `abi` need to have? Is it just the function, the entire contract, or something else?

Reply:
At a minimum:
- The function we're calling
- Any custom errors on the contract

Reply:
It's hard for me to tell from the PR. What validation / errors do we return if they botched it? 

This seems like a probably place to screw up

Reply:
Right now it'll return a `SendTransactionError` containing the viem error if they don't specify the ABI entry corresponding to the function name they are calling. It's unfortunately not really possible for us to check that they are including the ABI entries for errors, but any modern tool that exports ABIs should include them